### DISCUSSION TITLE
Unable to add a manually setup webradio favorite to the queue.

### DISCUSSION THREAD

**Discussion by dln949** [jcorporation/myMPD#908] [Q&A] [answered]
I manually set up a webradio favorite.  Then I select from the dropdown options "add to queue", but it does not get added to the queue.  Am I following the wrong steps?  

(If it matters, the station I'm attempting to add has the URI of http://ca.ah.fm/192k .)

**Comment by jcorporation** [comment]
This should work as you described. I need logs to help you further.

Edit: tested with your stream and it worked without flaws in my setup.

**Comment by dln949** [comment]
Well.....  It looks like logs will be a problem.  

When I follow the logging process here:  [https://jcorporation.github.io/myMPD/debug](url) , following the "myMPD logging" section does not work at all to give me any logs, when I get to the step to run mympd in the terminal all I see for output is this:  

mympd
ERROR    mympd     Work dir: creating "/var/lib/mympd" failed
ERROR    mympd     File exists

Not at all sure what that is trying to tell me.  

And when I follow the "webbrowser logging" section process, well, I can do all the steps shown, but no logging output appears anywhere that I can see per those brief instructions.  

I am running version 9.1.2-1ppa1~bionic of mympd on Ubuntu 18.04.

**Reply by jcorporation** [reply]
> I am running version 9.1.2-1ppa1~bionic of mympd on Ubuntu 18.04.

Please upgrade to latest version before reporting bugs.

Debug log:
You have started myMPD under a different user as it runs normal. Start myMPD as root, myMPD drops privileges by itself.

You can also set the log level in Settings->Maintenance to debug and get the output from your logging service, for systemd it is: `journalctl -fu mympd`

**Reply by dln949** [reply]
Thanks, I have the log to show you below.  First, a couple items I'm confused on.  

You wrote, "Please upgrade to latest version before reporting bugs."  Are you saying this is a bug?  Just to be clear, I was not reporting a bug - since I never used this application before, I didn't know if I was doing things correctly, I thought the problem was with my lack of understanding.  That's why I put this in the discussions area instead of the issues area.  Should I be moving this to the issues area?  

Also - How to I find the latest version that will work on Ubuntu 18.04?  I read the instructions found at [https://jcorporation.github.io/myMPD/installation/](url) , but it wasn't clear to me how I find the version that will work with Ubuntu 18.04.  (Also, the stuff about Docker and about compiling had me completely baffled.)  This is why I did the installation from the PPA.  Was that a bad decision?  


Here is the log, the messages that happened after attempting to add the webradio stations I set up to the queue start at the timestamp of 08:27:02:  

journalctl -fu mympd
-- Logs begin at Tue 2018-07-17 08:07:16 CDT. --
Nov 29 08:26:12 system1 mympd[16636]: DEBUG    webserver Sending response to conn_id 27 (length: 518): {"jsonrpc":"2.0","id":0,"result":{"method":"MYMPD_API_PLAYER_CURRENT_SONG","pos":4,"currentSongId":51,"Artist":["-"],"Album":"-","AlbumArtist":["-"],"Title":"The Temptations - Treat Her Like A Lady","Genre":["-"],"Disc":"-","Track":"-","Name":"FluxFM - Event 01","Duration":0,"LastModified":0,"uri":"http://streams.fluxfm.de/event01/mp3-320/streams.fluxfm.de/","stickerPlayCount":0,"stickerSkipCount":0,"stickerLike":1,"stickerLastPlayed":0,"stickerLastSkipped":0,"bookletPath":"","images": [],"startTime":1669731972}}
Nov 29 08:26:12 system1 mympd[16636]: DEBUG    webserver Sending 518 bytes to 27
Nov 29 08:26:13 system1 mympd[16636]: DEBUG    mympdapi  Leaving mpd idle mode
Nov 29 08:26:13 system1 mympd[16636]: DEBUG    mympdapi  Handle request
Nov 29 08:26:13 system1 mympd[16636]: INFO     mympdapi  MYMPD API request (26)(0) MYMPD_API_SETTINGS_GET: {"jsonrpc":"2.0","id":0,"method":"MYMPD_API_SETTINGS_GET","params":{}}
Nov 29 08:26:13 system1 mympd[16636]: DEBUG    mympdapi  Push response to web_server_queue for connection 26: {"jsonrpc":"2.0","id":0,"result":{"method":"MYMPD_API_SETTINGS_GET","mympdVersion":"9.1.2","mpdHost":"192.168.1.19","mpdPort":6602,"mpdPass":"dontsetpassword","mpdStreamPort":8080,"mpdTimeout":30000,"mpdKeepalive":false,"mpdBinarylimit":8192,"pin":false,"featCacert":true,"featScripting":true,"jukeboxMode":"off","coverimageNames":"folder,cover","jukeboxPlaylist":"Database","jukeboxQueueLength":1,"jukeboxUniqueTag":"Artist","jukeboxLastPlayed":24,"autoPlay":false,"loglevel":7,"smartpls":true,"smartplsSort":"","smartplsPrefix":"myMPDsmart","smartplsInterval":14400,"lastPlayedCount":200,"musicDirectory":"/media/main19/Toshiba/multimedia/MUSIC","playlistDirectory":"/var/lib/mpd/playlists","bookletName":"booklet.pdf","volumeMin":0,"volumeMax":100,"volumeStep":5,"lyricsUsltExt":"txt","lyricsSyltExt":"lrc","lyricsVorbisUslt":"LYRICS","lyricsVorbisSylt":"SYNCEDLYRICS","covercacheKeepDays":7,"colsQueueCurrent":["Pos","Title","Artist","Album","Du...
Nov 29 08:26:13 system1 mympd[16636]: DEBUG    mympdapi  Entering mpd idle mode
Nov 29 08:26:13 system1 mympd[16636]: DEBUG    webserver Got API response for id "26"
Nov 29 08:26:13 system1 mympd[16636]: DEBUG    webserver Sending response to conn_id 26 (length: 3339): {"jsonrpc":"2.0","id":0,"result":{"method":"MYMPD_API_SETTINGS_GET","mympdVersion":"9.1.2","mpdHost":"192.168.1.19","mpdPort":6602,"mpdPass":"dontsetpassword","mpdStreamPort":8080,"mpdTimeout":30000,"mpdKeepalive":false,"mpdBinarylimit":8192,"pin":false,"featCacert":true,"featScripting":true,"jukeboxMode":"off","coverimageNames":"folder,cover","jukeboxPlaylist":"Database","jukeboxQueueLength":1,"jukeboxUniqueTag":"Artist","jukeboxLastPlayed":24,"autoPlay":false,"loglevel":7,"smartpls":true,"smartplsSort":"","smartplsPrefix":"myMPDsmart","smartplsInterval":14400,"lastPlayedCount":200,"musicDirectory":"/media/main19/Toshiba/multimedia/MUSIC","playlistDirectory":"/var/lib/mpd/playlists","bookletName":"booklet.pdf","volumeMin":0,"volumeMax":100,"volumeStep":5,"lyricsUsltExt":"txt","lyricsSyltExt":"lrc","lyricsVorbisUslt":"LYRICS","lyricsVorbisSylt":"SYNCEDLYRICS","covercacheKeepDays":7,"colsQueueCurrent":["Pos","Title","Artist","Album","Duration...
Nov 29 08:26:13 system1 mympd[16636]: DEBUG    webserver Sending 3339 bytes to 26
Nov 29 08:26:37 system1 mympd[16636]: DEBUG    webserver WS message (29): ping
Nov 29 08:27:02 system1 mympd[16636]: DEBUG    webserver WS message (29): ping
Nov 29 08:27:04 system1 mympd[16636]: INFO     webserver HTTP request (26): POST /api/
Nov 29 08:27:04 system1 mympd[16636]: DEBUG    webserver API request (26): {"jsonrpc":"2.0","id":0,"method":"MYMPD_API_PLAYER_STATE","params":{}}
Nov 29 08:27:04 system1 mympd[16636]: INFO     webserver API request (26): MYMPD_API_PLAYER_STATE
Nov 29 08:27:04 system1 mympd[16636]: DEBUG    mympdapi  Leaving mpd idle mode
Nov 29 08:27:04 system1 mympd[16636]: DEBUG    mympdapi  Handle request
Nov 29 08:27:04 system1 mympd[16636]: INFO     mympdapi  MYMPD API request (26)(0) MYMPD_API_PLAYER_STATE: {"jsonrpc":"2.0","id":0,"method":"MYMPD_API_PLAYER_STATE","params":{}}
Nov 29 08:27:04 system1 mympd[16636]: DEBUG    mympdapi  Push response to web_server_queue for connection 26: {"jsonrpc":"2.0","id":0,"result":{"method":"MYMPD_API_PLAYER_STATE","state":"stop","volume":-1,"songPos":4,"elapsedTime":0,"totalTime":0,"currentSongId":51,"kbitrate":0,"queueLength":5,"queueVersion":482,"nextSongPos":-1,"nextSongId":-1,"lastSongId":-1,"AudioFormat":{"sampleRate":0,"bits":0,"channels":0},"lastError":""}}
Nov 29 08:27:04 system1 mympd[16636]: DEBUG    mympdapi  Entering mpd idle mode
Nov 29 08:27:04 system1 mympd[16636]: DEBUG    webserver Got API response for id "26"
Nov 29 08:27:04 system1 mympd[16636]: DEBUG    webserver Sending response to conn_id 26 (length: 322): {"jsonrpc":"2.0","id":0,"result":{"method":"MYMPD_API_PLAYER_STATE","state":"stop","volume":-1,"songPos":4,"elapsedTime":0,"totalTime":0,"currentSongId":51,"kbitrate":0,"queueLength":5,"queueVersion":482,"nextSongPos":-1,"nextSongId":-1,"lastSongId":-1,"AudioFormat":{"sampleRate":0,"bits":0,"channels":0},"lastError":""}}
Nov 29 08:27:04 system1 mympd[16636]: DEBUG    webserver Sending 322 bytes to 26
Nov 29 08:27:06 system1 mympd[16636]: INFO     webserver HTTP request (26): POST /api/
Nov 29 08:27:06 system1 mympd[16636]: DEBUG    webserver API request (26): {"jsonrpc":"2.0","id":0,"method":"MYMPD_API_QUEUE_APPEND_PLAYLIST","params":{"plist":"mympd://webradio/http___ca_ah_fm_192k.m3u","play":false}}
Nov 29 08:27:06 system1 mympd[16636]: INFO     webserver API request (26): MYMPD_API_QUEUE_APPEND_PLAYLIST
Nov 29 08:27:07 system1 mympd[16636]: DEBUG    mympdapi  Leaving mpd idle mode
Nov 29 08:27:07 system1 mympd[16636]: DEBUG    mympdapi  Handle request
Nov 29 08:27:07 system1 mympd[16636]: INFO     mympdapi  MYMPD API request (26)(0) MYMPD_API_QUEUE_APPEND_PLAYLIST: {"jsonrpc":"2.0","id":0,"method":"MYMPD_API_QUEUE_APPEND_PLAYLIST","params":{"plist":"mympd://webradio/http___ca_ah_fm_192k.m3u","play":false}}
Nov 29 08:27:07 system1 mympd[16636]: ERROR    mympdapi  Can not get list of inteface ip addresses
Nov 29 08:27:07 system1 mympd[16636]: ERROR    mympdapi  Address family not supported by protocol
Nov 29 08:27:07 system1 mympd[16636]: ERROR    mympdapi  MPD error: No such playlist (9)
Nov 29 08:27:07 system1 mympd[16636]: DEBUG    mympdapi  Setting interesting mpd tag types
Nov 29 08:27:07 system1 mympd[16636]: ERROR    mympdapi  Error in response to command mpd_run_load
Nov 29 08:27:07 system1 mympd[16636]: DEBUG    mympdapi  Push response to web_server_queue for connection 26: {"jsonrpc":"2.0","id":0,"error":{"method":"MYMPD_API_QUEUE_APPEND_PLAYLIST","facility":"mpd","severity":"error","message":"No such playlist","data":{}}}
Nov 29 08:27:07 system1 mympd[16636]: DEBUG    mympdapi  Entering mpd idle mode
Nov 29 08:27:07 system1 mympd[16636]: DEBUG    webserver Got API response for id "26"
Nov 29 08:27:07 system1 mympd[16636]: DEBUG    webserver Sending response to conn_id 26 (length: 152): {"jsonrpc":"2.0","id":0,"error":{"method":"MYMPD_API_QUEUE_APPEND_PLAYLIST","facility":"mpd","severity":"error","message":"No such playlist","data":{}}}
Nov 29 08:27:07 system1 mympd[16636]: DEBUG    webserver Sending 152 bytes to 26

**Comment by jcorporation** [comment, marked as answer]
I had better wrote "Always upgrade to latest version, older versions are not supported", but let me help you anyway.

The log shows the error:

```
Nov 29 08:27:07 system1 mympd[16636]: ERROR mympdapi Can not get list of inteface ip addresses
Nov 29 08:27:07 system1 mympd[16636]: ERROR mympdapi Address family not supported by protocol
Nov 29 08:27:07 system1 mympd[16636]: ERROR mympdapi MPD error: No such playlist (9)
```

You stumbled up on a bug that is fixed since v9.2.3: https://github.com/jcorporation/myMPD/issues/716

Ubuntu 18.04 is not supported by recent versions of myMPD, thats why there are no published official debs: https://download.opensuse.org/repositories/home:/jcorporation/

You should upgrade to a newer ubuntu version.

**Reply by dln949** [reply]
Ok, I understand the issues now.  Thanks for your help with this.
